scheduler/executor/schedule_checkpointer.go

```diff
@@ -0,0 +1,115 @@
+package executor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/flyteorg/flyteadmin/scheduler/executor/interfaces"
+	"github.com/flyteorg/flyteadmin/scheduler/repositories"
+	"github.com/flyteorg/flyteadmin/scheduler/repositories/models"
+	"github.com/flyteorg/flytestdlib/contextutils"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/flyteorg/flytestdlib/promutils"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"runtime/debug"
+	"runtime/pprof"
+	"sync"
+	"time"
+)
+
+
+const snapShotVersion = 1
+
+type checkPointerMetrics struct {
+	Scope                        promutils.Scope
+	CheckPointPanicCounter       prometheus.Counter
+	CheckPointSaveErrCounter     prometheus.Counter
+	CheckPointCreationErrCounter prometheus.Counter
+}
+
+type ScheduleCheckPointer struct {
```

Comment:
it is very confusing to see the interface and the struct name the same - ScheduleCheckPointer
Can we instead call it - DBSchedulerCheckpointer

Reply:
and add docs that say, that this checkpoints to a DB.

Also i would move the interfaces to a separate folder and this implementation to a separate folder. Everything in one folder seems very messy